HEAnalytics/HEAnalyticsPlatformLocalytics.swift

```diff
@@ -0,0 +1,163 @@
+//  and/or other materials provided with the distribution.
+//
+//  * Neither the name of HEAnalytics nor the names of its
+//  contributors may be used to endorse or promote products derived from
+//  this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+//  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+//  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+//  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+//  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+//  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+//  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+//  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import UIKit
+
+@objc(HEAnalyticsPlatformLocalytics)
+public class HEAnalyticsPlatformLocalytics: HEAnalyticsPlatform {
+    // Tracks internals state.
+    private var started = false
+    
+    /**
+     Initializes the platform with the given data.
+     
+     Subclasses are required to override and implement this in whatever way gets the platform's SDK intialized and ready (but not started). Think of it like starting the motor on the car and letting it idle being ready to go (whereas start() is when the car is put in gear and your foot depresses the gas pedal).
+     
+     Subclasses should invoke super (generally at the end, before returning).
+     
+     - parameter platformData: The platform's unique settings data, usually including whatever identifier/key is used to identify this app, and any other configuration data that may be relevant to the platform. The keys and values for each platform is unique to that platform.
+     */
+    internal override func initializePlatform(platformData: [NSObject:AnyObject]) {
+        let trackingID = platformData["trackingID"] as! String
+        Localytics.integrate(trackingID)
+        
+        if let dispatchInterval = platformData["dispatchInterval"] as? NSTimeInterval {
+            Localytics.setSessionTimeoutInterval(dispatchInterval)
+        }
+        else {
+            Localytics.setSessionTimeoutInterval(60) // Localytics's documented default.
+        }
+        
+        super.initializePlatform(platformData)
+    }
+    
+    
+    /// Has the user opt'd out of data collection? Note this value is not persisted anywhere by HEAnalytics. Exposing this setting in the GUI, persisting the value, restoring the value, and enforcing it generally is the responsibility of the app developer.
+    public override var optOut: Bool {
+        didSet {
+            Localytics.closeSession()
```

Comment:
What if this gets set to false (they don't want to optOut? maybe they opted out, changed their mind and want to opt back in)?